lib/spack/spack/relocate.py

```diff
@@ -360,17 +360,12 @@ def replace_prefix_text(path_name, old_dir, new_dir):
     Replace old install prefix with new install prefix
     in text files using utf-8 encoded strings.
     """
-
-    def replace(match):
-        return match.group().replace(old_dir.encode('utf-8'),
-                                     new_dir.encode('utf-8'))
     with open(path_name, 'rb+') as f:
         data = f.read()
         f.seek(0)
-        pat = re.compile(old_dir.encode('utf-8'))
-        if not pat.search(data):
-            return
-        ndata = pat.sub(replace, data)
+        pat = re.compile(rb'(?<![\w\-_/])[\w\-_]*?%s([\w\-_/]*)' % old_dir.encode('utf-8'))
```

Comment:
I don't believe that `rb` works in python 2.  Does it have to be `rb`?  Won't `r` work?

Reply:
Can you document this to explain what the negative lookbehind does?  If I understand this correctly, it's saying to *only* replace `old_dir` at the beginning of the path by ensuring that the string is *not* preceded by words, dashes, underscores, or a slash.  Is that basically right?

Reply:
`r` won't work in this context because we're operating on `byte` objects, so everything in `re` needs to be `byte` objects as well.

Reply:
I'll need to use `b` and just escape my backslashes.